### DISCUSSION TITLE
python/system dependencies

### DISCUSSION THREAD

**Discussion by lufinkey** [SteamDeckHomebrew/decky-loader#302] [General]
Is there an easy way to specify python and pacman dependencies for a plugin? I was going to include python dependencies as submodules and prompt the user for their password to install any pacman dependencies, but is there a better way to do this?

**Comment by TrainDoctor** [comment]
Decky assumes that the underlying filesystem is unmodified and we don't have any plans to support modification of the filesystem from decky plugins.
When it comes to python dependencies, unfortunately submodules are your best bet. Decky explicitly does not provide depdency resolution as it falls out of scope for Loader's goals. Those goals being allowing "plug and play" additive functionality for SteamOS game mode, and to allow developers freedom in their approach to adding functionality. We discussed dependency resolution very early on and we decided against it to keep our scope reasonable. Hopefully this answered your question.

**Reply by lufinkey** [reply]
Understandable! If I were to write support for python's `requirements.txt` for decky plugins at some point, would that be acceptable? Could even be as simple as a command that runs within the plugin template on  `pnpm run build`. It could just pull python dependencies from `requirements.txt` and puts them in a gitignored folder (`py_modules` or something?)

**Reply by lufinkey** [reply]
This might require PYTHONPATH to be appended to, potentially from decky loader itself
